File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/media-campaign/types/api/approval.ts
import type { MediaVersion } from './mediaVersion';

export interface Approval {
  approvalId: number;
  approvalStatus?: string;
  comment?: string;
  approvalDate: string; // ISO date string
  mediaTaskId: number;
  submittedMediaVersionId?: number;
  submittedMediaVersion?: MediaVersion;
}

File: MusicEventManagementSystem/MusicEventManagementSystem.API/Models/Approval.cs
namespace MusicEventManagementSystem.API.Models
{
    public class Approval
    {
        public int ApprovalId { get; set; }
        public string? ApprovalStatus { get; set; }
        public string? Comment { get; set; }
        public DateTime ApprovalDate { get; set; }
        public int MediaTaskId { get; set; }
        public int? SubmittedMediaVersionId { get; set; }

        public virtual MediaTask MediaTask { get; set; }
        public virtual MediaVersion? SubmittedMediaVersion { get; set; }

    }
}